src/compile/mark/mark.ts

```diff
@@ -36,7 +38,7 @@ export function parseMark(model: UnitModel): any[] {
 function parsePathMark(model: UnitModel) { // TODO: extract this into compilePathMark
   const mark = model.mark();
   // TODO: replace this with more general case for composition
-  const hasParentData = !!model.parent();
+  const isFaceted = model.parent() && model.parent().isFacet();
```

Comment:
Sure, we should at least put the right name in and add TODO for the splom and grand parent case.